OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -208,75 +208,75 @@ public interface INotifyHarvestAction
 		void MovementCancelled(Actor self);
 	}
 
-	public interface IDockClientInfo : ITraitInfoInterface { }
+	public interface ILinkClientInfo : ITraitInfoInterface { }
 
-	public interface IDockClient
+	public interface ILinkClient
 	{
-		BitSet<DockType> GetDockType { get; }
+		BitSet<LinkType> LinkType { get; }
 
 		/// <summary>When null, the client should act as if it can dock but never do.</summary>
-		DockClientManager DockClientManager { get; }
-		void OnDockStarted(Actor self, Actor hostActor, IDockHost host);
-		bool OnDockTick(Actor self, Actor hostActor, IDockHost dock);
-		void OnDockCompleted(Actor self, Actor hostActor, IDockHost host);
+		LinkClientManager LinkClientManager { get; }
+		void OnLinkCreated(Actor self, Actor hostActor, ILinkHost host);
+		bool OnLinkTick(Actor self, Actor hostActor, ILinkHost dock);
+		void OnLinkRemoved(Actor self, Actor hostActor, ILinkHost host);
 
 		/// <summary>Is this client allowed to dock.</summary>
 		/// <remarks>
-		/// Does not check if <see cref="Traits.DockClientManager"/> is enabled.
-		/// Function should only be called from within <see cref="IDockClient"/> or <see cref="Traits.DockClientManager"/>.
+		/// Does not check if <see cref="Traits.LinkClientManager"/> is enabled.
+		/// Function should only be called from within <see cref="ILinkClient"/> or <see cref="Traits.LinkClientManager"/>.
```

Comment:
Why is this exposed on the public interface if its only supposed to be used internally by the implementation?!?

Reply:
Because it needs to be implemented by clients and managers. How else would you expose it to all these users?